### PR TITLE
chapter 8 created

### PR DESCRIPTION
**Pull request by agustinaGallo97**
## :link: Jira Tracker reference:
*[AP - Chapter 8] UI Fragments and the Fragment Manager -
https://androidtraining.atlassian.net/browse/AT-44

## Description:
Chapter 8 created

## :camera_flash: UI preview
ezgif com-video-to-gif(7)

## :pushpin: Notes:
it also includes detekt integration

### CODE REVIEW

**Review by FerrariBruno — COMMENTED** [review]
Untrack all of the .idea files as well as the files under the /build and /config folders

File: .idea/codeStyles/Project.xml [hunk 1]
```diff
@@ -1,8 +1,5 @@
 <component name="ProjectCodeStyleConfiguration">
   <code_scheme name="Project" version="173">
-    <AndroidXmlCodeStyleSettings>
```
Comment (mirland) [inline .idea/codeStyles/Project.xml:3]: you shouldn't remove it

File: app/.gitignore [hunk 2]
```diff
@@ -10,7 +10,7 @@
 .externalNativeBuild
 
 .idea
```
Comment (mirland) [inline app/.gitignore]: it's wrong, you have not a .idea folder in the app folder

File: app/src/main/java/com/bignerdranch/android/criminalintent/MainActivity.kt [hunk 3]
```diff
@@ -2,11 +2,19 @@ package com.bignerdranch.android.criminalintent
...
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+
+        val currentFragment = supportFragmentManager.findFragmentById(R.id.fragmentContainer)
+        if (currentFragment == null) {
+            val fragment = CrimeFragment()
+            supportFragmentManager.beginTransaction().add(R.id.fragmentContainer, fragment).commit()
+        }
+
```
Comment (mirland) [inline app/src/main/java/com/bignerdranch/android/criminalintent/MainActivity.kt]:
```suggestion
```

File: app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeFragment.kt [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        val view = inflater.inflate(R.layout.fragment_crime, container, false)
+
+        titleField = view.findViewById(R.id.crimeTitle) as EditText
```
Comment (mirland) [inline app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeFragment.kt:29]: maybe a method to bind views?

File: app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeFragment.kt [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+        val titleWatcher = object : TextWatcher {
+            override fun afterTextChanged(s: Editable?) {}
+
+            override fun beforeTextChanged(
+                s: CharSequence?,
+                start: Int,
+                count: Int,
+                after: Int
+            ) {
```
Comment (mirland) [inline app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeFragment.kt]: one line?

File: app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeFragment.kt [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+            isEnabled = false
+        }
+
+        return view
+    }
+
+    override fun onStart() {
+        super.onStart()
+
+        val titleWatcher = object : TextWatcher {
```
Comment (mirland) [inline app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeFragment.kt:46]: use methods

File: app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeFragment.kt [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+            }
+
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                crime.title = s.toString()
+            }
+        }
+
+        titleField.addTextChangedListener(titleWatcher)
+
+        solvedCheckBox.apply {
```
Comment (mirland) [inline app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeFragment.kt:58]: why are you setting it in the onstart method?

File: app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeFragment.kt [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+        super.onCreate(savedInstanceState)
+        crime = Crime()
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        val view = inflater.inflate(R.layout.fragment_crime, container, false)
```
Comment (mirland) [inline app/src/main/java/com/bignerdranch/android/criminalintent/controllers/CrimeFragment.kt]: you can pass the id in the constructor, just a comment :smile:

File: app/src/main/res/layout/activity_main.xml [hunk 9]
```diff
@@ -1,18 +1,7 @@
...
-        android:layout_height="wrap_content"
-        android:text="Hello World!"
-        app:layout_constraintBottom_toBottomOf="parent"
-        app:layout_constraintLeft_toLeftOf="parent"
-        app:layout_constraintRight_toRightOf="parent"
-        app:layout_constraintTop_toTopOf="parent" />
-
-</androidx.constraintlayout.widget.ConstraintLayout>
\ No newline at end of file
+    />
```
Comment (mirland) [inline app/src/main/res/layout/activity_main.xml]: You can use a fragment container

Comment (mirland) [reply app/src/main/res/layout/activity_main.xml]: nl

File: app/src/main/res/layout/fragment_crime.xml [hunk 10]
```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (mirland) [inline app/src/main/res/layout/fragment_crime.xml]: nl

File: app/src/main/res/layout/fragment_crime.xml [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+        style="?android:listSeparatorTextViewStyle"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="@string/title_label" />
+
+    <EditText
+        android:id="@+id/crimeTitle"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:hint="@string/title_hint" />
```
Comment (mirland) [inline app/src/main/res/layout/fragment_crime.xml]: indentation is wrong

File: build/reports/detekt/detekt.html [hunk 12]
```diff
@@ -0,0 +1,162 @@
+<!DOCTYPE html>
```
Comment (mirland) [inline build/reports/detekt/detekt.html]: it shouldn't be tracked!

File: build/reports/detekt/detekt.txt [hunk 13]
```diff
@@ -0,0 +1,6 @@
+FinalNewline - [Element(kotlin.FILE)] at /home/toshiba/Desktop/CriminalIntent-Repo/CriminalIntentRepository/app/src/main/java/com/bignerdranch/android/criminalintent/models/Crime.kt:1:1 - Signature=
```
Comment (mirland) [inline build/reports/detekt/detekt.txt]: the build folder !

File: .gitignore [hunk 14]
```diff
@@ -1,23 +1,17 @@
...
-*.ear
-*.zip
-*.tar.gz
-*.rar
-
-# virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml
-hs_err_pid*
+build
+/reports
+/secrets
```
Comment (mirland) [inline .gitignore]:
```suggestion
/secrets

```

File: app/src/main/res/layout/fragment_crime.xml [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+        android:id="@+id/crimeDate"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        tools:text="Wed Nov 14 11:56 EST 2018" />
+
+    <CheckBox
+        android:id="@+id/crimeSolved"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="@string/solved_label" />
```
Comment (mirland) [inline app/src/main/res/layout/fragment_crime.xml]: indentation

**Review by mirland — APPROVED** [review]

**Review by FerrariBruno — COMMENTED** [review]
Add Feli to the review :)

**Review by f7deleon — APPROVED** [review]

**Pull request merged**